File: frontend/src/api/preferences.ts
import { api } from './client'

export interface PlanPreferences {
  default_formula: 'mifflin' | 'harris' | 'schofield'
  default_activity_level: number
  default_goal: 'cut' | 'maintenance' | 'bulk'
  protein_pct: number
  fat_pct: number
  carb_pct: number
  kcal_adjustment_cut: number
  kcal_adjustment_bulk: number
}

export async function getPlanPreferences() {
  const { data } = await api.get<PlanPreferences>('/plan-preferences')
  return data
}

export async function savePlanPreferences(payload: PlanPreferences) {
  const { data } = await api.put<PlanPreferences>('/plan-preferences', payload)
  return data
}

File: backend/schemas/preferences.py
from pydantic import BaseModel
from typing import Literal, Optional


class PlanPreferencesUpdate(BaseModel):
    default_formula: Literal["mifflin", "harris", "schofield"] = "mifflin"
    default_activity_level: float = 1.55
    default_goal: Literal["cut", "maintenance", "bulk"] = "cut"
    protein_pct: float = 25
    fat_pct: float = 20
    carb_pct: float = 55
    kcal_adjustment_cut: float = -300
    kcal_adjustment_bulk: float = 300


class PlanPreferencesResponse(PlanPreferencesUpdate):
    class Config:
        from_attributes = True
